Contribute/how-to-write-use-markdown.md

```diff
@@ -383,10 +383,329 @@ You can see an example of selectors in action at the [Azure docs](https://docs.m
+
+Prefer snippets over hard-coded line numbers, as explained in the next section.
+
+#### Named snippet
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" id="snippet_Create":::
+```
+
+Use only letters and underscores for the name.
+
+The example displays the `snippet_Create` section of the code file. The code file for this example
+has a C# region named `snippet_Create`:
+
+```cs
+// code excluded from the snippet
+// <snippet_Create>
+// code included in the snippet
+// </snippet_Create>
+// code excluded from the snippet
+```
+
+Whenever you can, refer to a named section rather than specifying line numbers. Line number
+references are brittle because code files inevitably change in ways that make line numbers change.
+You don't necessarily get notified of such changes. Your article eventually starts showing the wrong
+lines and you have no clue anything has changed.
+
+#### Highlighting selected lines
+
+Example:
+
+```markdown
+:::code language="csharp" source="intro/samples/cu/Controllers/StudentsController.cs" range="2-24,26" highlight="2,5":::
+```
+
+The example highlights lines 2 and 5, counting from the start of the displayed snippet. (Line
+numbers to highlight don't count from the start of the code file.) In other words, lines 3 and 6
+of the code file are highlighted.
+
+#### Interactive code snippets
+
+You can enable interactive mode for code snippets included by reference. Here are examples:
+
+```markdown
+:::code source="PowerShell.ps1" interactive="cloudshell-powershell":::
+```
+
+```markdown
+:::code source="Bash.sh" interactive="cloudshell-bash":::
+```
+
+To turn on this feature for a particular code block, use the `interactive` attribute. The
+available attribute values are:
+
+- `cloudshell-powershell` - Enables the Azure PowerShell Cloud Shell, as in the preceding
+  example
+- `cloudshell-bash` - Enables the Azure Cloud Shell
+- `try-dotnet` - Enables Try .NET
+- `try-dotnet-class` - Enables Try .NET with class scaffolding
+- `try-dotnet-method` - Enables Try .NET with method scaffolding
+
+For the Azure Cloud Shell and PowerShell Cloud Shell, users can run commands against only their own
+Azure account.
+
+[Try .NET](https://github.com/dotnet/try) enables interactive execution of .NET code (C#) in the browser. For Try .NET, there are three options for interactivity: `try-dotnet`, `try-dotnet-class`, and `try-dotnet-method`. Use of these options require no additional configuration within the code snippet. The namespaces currently available by default are:
+
+- System
+- System.Linq
+- System.Collections.Generic
+- System.Text
+- System.Globalization
+- System.Text.RegularExpressions
+
+The `try-dotnet` attribute value enables users to run C# code in the browser without the need to wrap the code in any custom code.
+
+Example:
+
+```md
+:::code source="relative/path/source.cs" interactive="try-dotnet"::
```

Comment:
don't you need to set the language here? or is it optional? 

also is it correct to have 3 ::: before and 2:: after?

Reply:
@mairaw thanks for catching the missing ':' - I'll add that in.
language isn't a required attribute given that it's derived from the source file extension. if you're using the Docs Authoring Pack, language populates after selecting the source.

Reply:
so as not to confuse anyone, I'll add language to the snippet examples